### PR TITLE
9064 Added typing to layouts.py

### LINKED ISSUE #9064: Add static typing to bokeh
This is a meta issue for adding static typing to bokeh:

- [x] `bokeh.palettes`
- [ ] `bokeh.embed.standalone`
- [ ] `bokeh.util.string` @mborus

### PR DESCRIPTION
**Pull request by srf94**
All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [ ] issues: addresses #9064

### CODE REVIEW

File: bokeh/layouts.py [hunk 1]
```diff
@@ -49,7 +51,11 @@
 # General API
 #-----------------------------------------------------------------------------
 
-def row(*args, **kwargs):
+
+def row(*args: LayoutDOM,
+        sizing_mode: Optional[LayoutDOM.sizing_mode] = None,
```
Comment (mattpap) [inline bokeh/layouts.py]: `LayoutDOM.sizing_mode` should be replaced with `SizingMode` enum from `bokeh.core.enums`.

Comment (srf94) [reply bokeh/layouts.py]: Changed as discussed

**Comment by srf94** [comment]
Currently failing mypy in two places.
Line 502 - container(list(map(traverse_dom, item.children)))
Line 517 - "List" has no attribute "__iter__" (not iterable)

The second error will be fixed when List becomes typed.

Otherwise, there are quite a few 'Any' types that can probably be tightened.

**Comment by bryevdv** [comment]
Going to go ahead and close this, @mattpap I'll leave the branch for now if you want it kept around.

**Pull request closed without merging**